src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -212,10 +219,12 @@ private static void _BulkMoveWithWriteBarrier(ref byte destination, ref byte sou
                 {
                     byteCount -= BulkMoveWithWriteBarrierChunk;
                     __BulkMoveWithWriteBarrier(ref Unsafe.AddByteOffset(ref destination, byteCount), ref Unsafe.AddByteOffset(ref source, byteCount), BulkMoveWithWriteBarrierChunk);
+                    Thread.FastPollGC();
                 }
                 while (byteCount > BulkMoveWithWriteBarrierChunk);
             }
             __BulkMoveWithWriteBarrier(ref destination, ref source, byteCount);
```

Comment:
Perhaps this can be renamed to `InternalBulkMoveWithWriteBarrier` or something which matches current FCall naming convention.

Reply:
This feels like pulling on a thread. This API is defined in both coreclr and native AOT and there are others in the same area that follow the same pattern. I'm happy to do this, but it is going to be very narrow.